app/controllers/books_controller.rb

```diff
@@ -1,62 +1,68 @@
 class BooksController < ApplicationController
-  def index
-    @books = Book.all
-  end
-
-  def show
-    @book = Book.find(params[:id])
-  end
-
-  def new
-    @book = Book.new
-    @subjects = Subject.all
-  end
-
-  def book_params
-    params.require(:books).permit(:title, :price, :subject_id, :description)
-  end
-
-  def create
-    @book = Book.new(book_params)
-    # Allow for empty subject, default to Physics
-    if @book.subject.nil?
-      @book.subject_id = 5 # default to Geography if no subject is given
-    end
-    if @book.save
-      redirect_to books_path
-    else
-      @subjects = Subject.all
-      flash[:errors] = @book.errors.full_messages
-      render action: "new"
-    end
-  end
-
-  def edit
-    @book = Book.find(params[:id])
-    @subjects = Subject.all
-  end
-
-  def book_param
-    params.require(:book).permit(:title, :price, :subject_id, :description)
-  end
-
-  def update
-    @book = Book.find(params[:id])
-
-    if @book.update_attributes(book_param)
-      redirect_to action: "show", id: @book
-    else
-      @subjects = Subject.all
-      render action: "edit"
-    end
-  end
-
-  def destroy
-    Book.find(params[:id]).destroy
-    redirect_to action: "index"
-  end
-
-  def show_subjects
-    @subject = Subject.find(params[:id])
-  end
+    
+    def index
+        @books=Book.all
+    end
+
+    def show
+        @book=Book.find(params[:id])
+    end
+
+    def new
+        @book=Book.new
+        @subjects=Subject.all
+        if params[:subject_id]
+            @book.subject_id=params[:subject_id]
+        end
+    end
+  
+    def create
+        @book = Book.new(book_param)
+#Allow for empty subject, default to "No Subject"
+        if @book.subject.nil?
+           @book.subject_id=6#ID for "No Subject"
```

Comment:
Hard-coding IDs isn't the best. What happens when it is 7 in production?

```suggestion
           @book.subject = Subject.find_by(name: "No Subject")
```